File: scala/src/main/scala/folio/codinginterview/application/repository/AccountRepository.scala
package folio.codinginterview.application.repository

import folio.codinginterview.domain.Account
import folio.codinginterview.domain.UserId
import scala.concurrent.Future

/** 口座管理リポジトリ。 */
trait AccountRepository {
  def find(userId: UserId): Future[Option[Account]]
  def upsert(userId: UserId, account: Account): Future[Unit]
  def exists(userId: UserId): Future[Boolean]
}

File: golang/internal/application/repository/account_repository.go
package repository

import "folio/codinginterview/internal/domain"

// AccountRepository は口座管理のリポジトリインターフェースです。
type AccountRepository interface {
	Find(userId domain.UserId) (*domain.Account, error)
	Upsert(userId domain.UserId, account domain.Account) error
	Exists(userId domain.UserId) (bool, error)
}
